nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/http/StandardHttpResponseMerger.java

```diff
@@ -0,0 +1,164 @@
+    private static final List<EndpointResponseMerger> endpointMergers = new ArrayList<>();
+
+    static {
+        endpointMergers.add(new ControllerStatusEndpointMerger());
+        endpointMergers.add(new GroupStatusEndpointMerger());
+        endpointMergers.add(new ProcessorStatusEndpointMerger());
+        endpointMergers.add(new ConnectionStatusEndpiontMerger());
+        endpointMergers.add(new PortStatusEndpointMerger());
+        endpointMergers.add(new RemoteProcessGroupStatusEndpointMerger());
+        endpointMergers.add(new ProcessorEndpointMerger());
+        endpointMergers.add(new ProcessorsEndpointMerger());
+        endpointMergers.add(new RemoteProcessGroupEndpointMerger());
+        endpointMergers.add(new RemoteProcessGroupsEndpointMerger());
+        endpointMergers.add(new ProcessGroupEndpointMerger());
+        endpointMergers.add(new FlowSnippetEndpointMerger());
+        endpointMergers.add(new ProvenanceQueryEndpointMerger());
+        endpointMergers.add(new ProvenanceEventEndpointMerger());
+        endpointMergers.add(new ControllerServiceEndpointMerger());
+        endpointMergers.add(new ControllerServicesEndpointMerger());
+        endpointMergers.add(new ControllerServiceReferenceEndpointMerger());
+        endpointMergers.add(new ReportingTaskEndpointMerger());
+        endpointMergers.add(new ReportingTasksEndpointMerger());
+        endpointMergers.add(new DropRequestEndpiontMerger());
+        endpointMergers.add(new ListFlowFilesEndpointMerger());
+        endpointMergers.add(new ComponentStateEndpointMerger());
+        endpointMergers.add(new BulletinBoardEndpointMerger());
+        endpointMergers.add(new StatusHistoryEndpointMerger());
+        endpointMergers.add(new SystemDiagnosticsEndpointMerger());
+        endpointMergers.add(new CountersEndpointMerger());
+    }
+
+    @Override
+    public NodeResponse mergeResponses(final URI uri, final String httpMethod, final Set<NodeResponse> nodeResponses) {
+        final boolean hasSuccess = hasSuccessfulResponse(nodeResponses);
+        if (!hasSuccess) {
+            // It doesn't really matter which response we choose, as all are problematic.
+            final NodeResponse clientResponse = nodeResponses.iterator().next();
+
+            // Drain the response from all nodes except for the 'chosen one'. This ensures that we don't
+            // leave data lingering on the socket and ensures that we don't consume the content of the response
+            // that we intend to respond with
+            drainResponses(nodeResponses, clientResponse);
+            return clientResponse;
+        }
+
+        // Determine which responses are successful
+        final Set<NodeResponse> successResponses = nodeResponses.stream().filter(p -> p.is2xx()).collect(Collectors.toSet());
+        final Set<NodeResponse> problematicResponses = nodeResponses.stream().filter(p -> !p.is2xx()).collect(Collectors.toSet());
```

Comment:
Consider moving it down after determining a merger since it is not used unless merger != null.